cluniverse/Luniverse.cs

```diff
@@ -0,0 +1,384 @@
+                        {
+                            msg = $"The deal is for a different client. Expected {ethSrcAddress}, got {srcAddress.Value}";
+                            return false;
+                        }
+                    }
+
+                    BigInteger transferAmount;
+                    if (!BigInteger.TryParse(amountString, out transferAmount) || transferAmount <= 0)
+                    {
+                        msg = "Invalid amount";
+                        return false;
+                    }
+
+                    BigInteger gain;
+                    if (!BigInteger.TryParse(gainString, out gain))
+                    {
+                        msg = "Invalid gain";
+                        return false;
+                    }
+
+                    transferAmount = transferAmount + gain;
+                    
+                    if (transferAmount < 0)
+                    {
+                        msg = "Overflow";
+                        return false;
+                    }
+
+                    var txCount = web3.Eth.Transactions.GetTransactionCount.SendRequestAsync(ethSrcAddress).Result;
+                    TransactionSigner signer = new TransactionSigner();
+
+                    HexBigInteger gasPrice;
+
+                    string gasPriceInGweiString = cfg["gasPriceInGwei"];
+                    if (int.TryParse(gasPriceInGweiString, out int gasPriceOverride))
+                    {
+                        gasPrice = new HexBigInteger(Nethereum.Util.UnitConversion.Convert.ToWei(gasPriceOverride, Nethereum.Util.UnitConversion.EthUnit.Gwei));
+                    }
+                    else
+                    {
+                        gasPrice = web3.Eth.GasPrice.SendRequestAsync().Result;
+                    }
+
+                    string to;
+                    string data;
+                    BigInteger amount;
+                    HexBigInteger gasLimit;
+                    if (erc20)
+                    {
+                        string creditcoinContract = cfg["creditcoinContract"];
+                        string creditcoinContractAbi = cfg["creditcoinContractAbi"];
+
+                        to = creditcoinContract;
+
+                        var contract = web3.Eth.GetContract(creditcoinContractAbi, creditcoinContract);
+                        var burn = contract.GetFunction("exchange");
+                        var functionInput = new object[] { transferAmount, txBuilder.getSighash() };
+                        data = burn.GetData(functionInput);
+                        gasLimit = burn.EstimateGasAsync(functionInput).Result;
+                        amount = 0;
+                    }
+                    else
+                    {
+                        gasLimit = web3.Eth.Transactions.EstimateGas.SendRequestAsync(new Nethereum.RPC.Eth.DTOs.CallInput(orderId, ethDstAddress, new Nethereum.Hex.HexTypes.HexBigInteger(transferAmount))).Result;
```

Comment:
Add an additional 10% on top of the estimated gas to ensure transaction fulfillment